### PR TITLE
refactor: migrate QueryAndSaveBtns to TypeScript and add stories

### PR DESCRIPTION
**Pull request by ad-m**
### SUMMARY

Migrated QueryAndSaveBtns to TypeScript to apply direction outlined in #18100 .

In addition, added a new story to visualize that component.
 
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

obraz

### TESTING INSTRUCTIONS

Storybook can be used to verify that changes.

### ADDITIONAL INFORMATION

- [x] Has associated issue: see #18100
- [x] Required feature flags: no
- [x] Changes UI: no
- [x] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351)): no
  - [x] Migration is atomic, supports rollback & is backwards-compatible: N/A
  - [x] Confirm DB migration upgrade and downgrade tested: N/A
  - [x] Runtime estimates and downtime expectations provided: N/A
- [x] Introduces new feature or API: no
- [x] Removes existing feature or API: no

@kgabryje , @asadUllah58 You might be interested in this PR!

### CODE REVIEW

File: superset-frontend/src/explore/components/QueryAndSaveBtns.tsx [hunk 1]
```diff
@@ -17,42 +17,37 @@
...
+  const {
+    canAdd,
+    onQuery = () => {},
+    onSave = () => {},
+    onStop = () => {},
+    loading,
+    chartIsStale,
+    errorMessage,
+  } = props;
+  let qryButtonStyle = 'tertiary' as ButtonStyle;
```
Comment (kgabryje) [inline superset-frontend/src/explore/components/QueryAndSaveBtns.tsx:46]:
```suggestion
  let qryButtonStyle: ButtonStyle = 'tertiary';
```

Comment (kgabryje) [reply superset-frontend/src/explore/components/QueryAndSaveBtns.tsx:46]: That will allow to remove the `as ButtonStyle` statements from lines 48 and 50

Comment (ad-m) [reply superset-frontend/src/explore/components/QueryAndSaveBtns.tsx]: Great idea. That notation in that situation is better. I applied that as a separate commit (Is squash necessary from me?).

**Comment by ad-m** [comment]
@kgabryje , CI finally fixed. Could you take a look?

**Review by geido — APPROVED** [review]
LGTM!

**Pull request merged**